src/background/services/tabState.ts

```diff
@@ -14,7 +13,7 @@ interface SaveOverpayingDetails {
 }
 
 export class TabState {
-  private state = new WeakMap<Tabs.Tab, Map<string, State>>()
+  private state = new Map<number, Map<string, State>>()
```

Comment:
So we're clear what the number represents.
Also consider providing same for `string` (as `SessionId`, but not relevant to this PR)

```suggestion
  type TabId = NonNullable<Tabs.Tab['id']>
  // ...
  private state = new Map<TabId, Map<string, State>>()
```